test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+
+    describe('find', function () {
+        describe('find function', function () {
+            it('should find Helen', function () {
+                var testArray = _.fill(new Array(50), 1).map(function () {
+                    if (Math.random() < 0.2) {
+                        return {name: 'Kodi', age: 12, active: true};
+                    } else if (Math.random() < 0.4) {
+                        return {name: 'Olaf', age: 32, active: false};
+                    } else if (Math.random() < 0.6) {
+                        return {name: 'Helen', age: 65, active: true};
+                    } else if (Math.random() < 0.8) {
+                        return {name: 'Pluto', age: 25, active: false};
+                    } else {
+                        return {name: 'Rudi', age: 43, active: true};
+                    }
+                });
+                var params = datasets.find(testArray);
+                var array = params[0];
+                var predicate = params[1];
+                var result = _.find.apply(_, params);
+
+                expect(array.length).to.eql(50);
+                expect(Object.keys(predicate).length).to.eql(1);
+                expect(typeof predicate.age).to.eql('number');
+                expect(_.values(result)).to.contain('Helen');
+                expect(_.values(result)).to.contain(65);
+            });
+        });
+        describe('find2 function', function () {
+            var params = datasets.find2();
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = typeof params[1];
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql('string');
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of objects', function () {
+                var array = params[0];
+
+                expect(array.length).to.eql(3);
+                array.forEach(function (n) {
+                    expect(_.has(n, 'NAME')).to.eql(true);
+                    expect(_.has(n, 'active')).to.eql(true);
+                    expect(_.has(n, 'age')).to.eql(true);
+
+                    if (typeof n.NAME !== 'string') {
+                        expect(typeof n.NAME).to.eql('string');
+                    } else if (typeof n.age !== 'number') {
+                        expect(typeof n.age).to.eql('number');
+                    } else if (typeof n.active !== 'boolean') {
+                        expect(typeof n.active).to.eql('boolean');
+                    }
+                });
+            });
+            it('should return first match for name predicate', function () {
+                expect(_.find.apply(_, datasets.find2())).to.eql({
+                    NAME: 'Rudi',
+                    age: 15,
+                    active: true,
+                    gender: 'M'
+                });
+            });
+        });
+    });
+
```

Comment:
We are missing function and array iteratee types for find. Moreover, we should provide some examples with a object as collection.